projects/admin/src/app/menu/menu.component.ts

```diff
@@ -85,18 +96,28 @@ export class MenuComponent implements OnInit {
         data.active = true;
         this.activeLanguagesMenuItem = data;
       }
-      this.languagesMenu.entries[1].entries.splice(0, 0, data);
+      this.languagesMenu.entries[0].entries.splice(0, 0, data);
     }
     this.userMenu.entries.push({
       name: `${currentUser.first_name[0]}${currentUser.last_name[0]}`,
       entries: [
         {
+          name: this.translateService.instant('Switch to public view'),
+          href: '/',
+          iconCssClass: 'fa fa-television'
+        }, {
           name: this.translateService.instant('Logout'),
           href: `/logout`,
           iconCssClass: 'fa fa-sign-out'
         }
       ]
     });
+
+    this.librarySwitchService.onGenerate$.subscribe((entries: any) => {
+      this.librariesSwitchMenuEntriesLength = entries.length;
```

Comment:
is it really useful, can probably replaced by a get function on `this.librariesSwitchMenu.entries.length`

Reply:
done